pkg/operator/ceph/csi/spec.go

```diff
@@ -470,18 +481,75 @@ func startDrivers(namespace string, clientset kubernetes.Interface, ver *version
 	}
 
 	if ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor >= provDeploymentSuppVersion) {
-		err = createCSIDriverInfo(clientset, RBDDriverName, ownerRef)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create CSI driver object for %q", RBDDriverName)
+		if EnableRBD {
+			err = createCSIDriverInfo(clientset, RBDDriverName, ownerRef)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI driver object for %q", RBDDriverName)
+			}
 		}
-		err = createCSIDriverInfo(clientset, CephFSDriverName, ownerRef)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create CSI driver object for %q", CephFSDriverName)
+		if EnableCephFS {
+			err = createCSIDriverInfo(clientset, CephFSDriverName, ownerRef)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI driver object for %q", CephFSDriverName)
+			}
+		}
+	}
+	return nil
+}
+
+func stopDrivers(namespace string, clientset kubernetes.Interface) error {
+	if !EnableRBD {
+		logger.Info("CSI Ceph RBD driver disabled ")
+		succeeded := deleteCSIDriverResources(namespace, clientset, RBDPlugin, RBDProvisioner, "csi-rbdplugin-metrics", RBDDriverName)
+		if succeeded {
+			logger.Info("successfully removed CSI Ceph RBD driver")
+		} else {
+			logger.Info("failed to remove CSI Ceph RBD driver")
+		}
+	}
+
+	if !EnableCephFS {
+		logger.Info("CSI CephFS driver disabled")
+		succeeded := deleteCSIDriverResources(namespace, clientset, CephFSPlugin, CephFSProvisioner, "csi-cephfsplugin-metrics", CephFSDriverName)
+		if succeeded {
+			logger.Info("successfully removed CSI CephFS driver")
+		} else {
+			logger.Info("failed to remove CSI CephFS driver")
 		}
 	}
 	return nil
 }
 
+func deleteCSIDriverResources(
+	namespace string, clientset kubernetes.Interface, daemonset string, deployment string, service string, driverName string) bool {
+
+	succeeded := true
+	err := k8sutil.DeleteDaemonset(clientset, namespace, daemonset)
+	if err != nil {
+		logger.Errorf("failed to delete the %s. %v", daemonset, err)
+		succeeded = false
+	}
+
+	err = k8sutil.DeleteDeployment(clientset, namespace, deployment)
```

Comment:
if the kubernetes version is 1.13.x the csi driver will be deployed as statefulset, based on the kubernetes version you need to delete the resource
1.13.x -> Delete Statefulset
1.14+ -> Delete Deployment